src/frequenz/channels/_latest_value_cache.py

```diff
@@ -32,31 +40,84 @@
 ```
 """
 
+from __future__ import annotations
+
 import asyncio
 import typing
 
 from ._receiver import Receiver
 
 T_co = typing.TypeVar("T_co", covariant=True)
+HashableT = typing.TypeVar("HashableT", bound=typing.Hashable)
 
 
-class _Sentinel:
+class Sentinel:
     """A sentinel to denote that no value has been received yet."""
 
+    def __init__(self, desc: str) -> None:
+        """Initialize the sentinel."""
+        self._desc = desc
+
     def __str__(self) -> str:
         """Return a string representation of this sentinel."""
-        return "<no value received yet>"
+        return f"<Sentinel: {self._desc}>"
 
 
-class LatestValueCache(typing.Generic[T_co]):
+NO_KEY = Sentinel("no key provided")
+NO_KEY_FUNCTION = Sentinel("no key function provided")
+NO_VALUE_RECEIVED = Sentinel("no value received yet")
```

Comment:
Nitpick:
```suggestion
NO_KEY: Final[Sentinel] = Sentinel("no key provided")
NO_KEY_FUNCTION: Final[Sentinel] = Sentinel("no key function provided")
NO_VALUE_RECEIVED: Final[Sentinel] = Sentinel("no value received yet")
```

But nice, it makes it much more readable indeed :tada:

Reply:
done